airflow/configuration.py

```diff
@@ -530,6 +530,24 @@ def parameterized_config(template):
 
 conf.read(AIRFLOW_CONFIG)
 
+if conf.has_option('core', 'AIRFLOW_HOME'):
+    if 'AIRFLOW_HOME' in os.environ:
+        warnings.warn(
+            'Specifying both AIRFLOW_HOME environment variable and airflow_home '
+            'in the config file is deprecated. Please use only the AIRFLOW_HOME '
+            'environment variable and remove the config file entry.',
+            category=DeprecationWarning,
+        )
+    else:
+        AIRFLOW_HOME = conf.get('core', 'airflow_home')
```

Comment:
What's the purpose of having this line?

Reply:
Double-checking on this: so your intention is that if the user is still using `airflow_home` in .cfg file, only a warning will be given (no hard stoping) and the value in .cfg file will be used eventually?

Reply:
Yes, I was trying to make this change as easy as possible - warn but carry on using the setting (for 99% of people it will be the same)

Though the 1 case I know that this will break is the Puckel docker container. So perhaps if the two settings are different I should throw an exception.

Reply:
Or throw an exception once `conf.has_option('core', 'AIRFLOW_HOME') == True`?

Personally I think it's better to make it as explicit as possible, and having a clean cut.

Reply:
Throwing an exception in this case would be a breaking change (as this was in the default template, so every install would and up stopping working) which I am trying to avoid.

Reply:
In this case, it’s not really necessary to have an if-else here to check whether AIRFLOW_HOME is in env? The two warning messages are not that different to me.

Reply:
Hi @ashb , I had a double-check on this. Let me correct myself for this:
- I agree it's necessary to have the if-else `if 'AIRFLOW_HOME' in os.environ:` inside `if conf.has_option('core', 'AIRFLOW_HOME'):`, and decide if `AIRFLOW_HOME` should be based on the value in .cfg file.
- But I don't think it's necessary to have the warning messages for two times in the code. The two warning messages here are very similar to each other, and either of them will be invoked anyway. We can have something like

```python
if conf.has_option('core', 'AIRFLOW_HOME'):
    warnings.warn('<warning msg>', category=DeprecationWarning,)

    if 'AIRFLOW_HOME' not in os.environ:
        AIRFLOW_HOME = conf.get('core', 'airflow_home')

```

Reply:
I've changed it to:

```python
if conf.has_option('core', 'AIRFLOW_HOME'):
    msg = (
        'Specifying both AIRFLOW_HOME environment variable and airflow_home '
        'in the config file is deprecated. Please use only the AIRFLOW_HOME '
        'environment variable and remove the config file entry.'
    )
    if 'AIRFLOW_HOME' in os.environ:
        warnings.warn(msg, category=DeprecationWarning,)
    elif conf.get('core', 'airflow_home') == AIRFLOW_HOME:
        warnings.warn(
            'Specifying airflow_home in the config file is deprecated. As you '
            'have left it at the default value you should remove the setting '
            'from your airflow.cfg and suffer no change in behaviour.',
            category=DeprecationWarning,
        )
    else:
        AIRFLOW_HOME = conf.get('core', 'airflow_home')
        warnings.warn(msg, category=DeprecationWarning,)

```

I   thought about adding an extra case do say which value it was